### ISSUE TITLE
unable to fold certain constants

### ISSUE THREAD

**Issue by ninono12345** [microsoft/onnxruntime#19249]
### Describe the issue

When I run my model through this command:

polygraphy surgeon sanitize tomp101_head_latest3.onnx --fold-constants -o latest3_sanitized2.onnx

I get this:

2024-01-16 17:44:05.4688013 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_12
2024-01-16 17:44:05.4766621 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_11
2024-01-16 17:44:05.4845569 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_10
2024-01-16 17:44:05.4931583 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_8
2024-01-16 17:44:05.5018787 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_6
2024-01-16 17:44:05.5097960 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_5
2024-01-16 17:44:05.5172133 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_4
2024-01-16 17:44:05.5234376 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_3
2024-01-16 17:44:05.5311036 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_2
2024-01-16 17:44:05.5372898 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze_1
2024-01-16 17:44:05.5443412 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /classifier/Unsqueeze
2024-01-16 17:44:05.5510944 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_32
2024-01-16 17:44:05.5568449 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_31
2024-01-16 17:44:05.5639316 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_30
2024-01-16 17:44:05.5698841 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_29
2024-01-16 17:44:05.5765543 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_28
2024-01-16 17:44:05.5828426 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze_17
2024-01-16 17:44:05.5902849 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze_16
2024-01-16 17:44:05.5986004 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze_15
2024-01-16 17:44:05.6057177 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze_14
2024-01-16 17:44:05.6143859 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze_9
2024-01-16 17:44:05.6214617 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/multihead_attn/Unsqueeze
2024-01-16 17:44:05.6299710 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_13
2024-01-16 17:44:05.6378020 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_12
2024-01-16 17:44:05.6459786 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_11
2024-01-16 17:44:05.6543992 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_10
2024-01-16 17:44:05.6623247 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_6
2024-01-16 17:44:05.6711435 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze_3
2024-01-16 17:44:05.6795708 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.5/self_attn/Unsqueeze
2024-01-16 17:44:05.6889410 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze_17
2024-01-16 17:44:05.6965514 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze_16
2024-01-16 17:44:05.7049768 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze_15
2024-01-16 17:44:05.7122874 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze_14
2024-01-16 17:44:05.7206417 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze_9
2024-01-16 17:44:05.7277411 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/multihead_attn/Unsqueeze
2024-01-16 17:44:05.7353374 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_13
2024-01-16 17:44:05.7433410 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_12
2024-01-16 17:44:05.7507542 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_11
2024-01-16 17:44:05.7586551 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_10
2024-01-16 17:44:05.7658402 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_6
2024-01-16 17:44:05.7741140 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze_3
2024-01-16 17:44:05.7818602 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.4/self_attn/Unsqueeze
2024-01-16 17:44:05.7896626 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze_17
2024-01-16 17:44:05.7972461 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze_16
2024-01-16 17:44:05.8056821 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze_15
2024-01-16 17:44:05.8142432 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze_14
2024-01-16 17:44:05.8214699 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze_9
2024-01-16 17:44:05.8301582 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/multihead_attn/Unsqueeze
2024-01-16 17:44:05.8378422 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_13
2024-01-16 17:44:05.8462113 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_12
2024-01-16 17:44:05.8543821 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_11
2024-01-16 17:44:05.8623293 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_10
2024-01-16 17:44:05.8711495 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_6
2024-01-16 17:44:05.8795980 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze_3
2024-01-16 17:44:05.8888297 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.3/self_attn/Unsqueeze
2024-01-16 17:44:05.8960275 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze_17
2024-01-16 17:44:05.9042903 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze_16
2024-01-16 17:44:05.9115070 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze_15
2024-01-16 17:44:05.9206712 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze_14
2024-01-16 17:44:05.9278957 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze_9
2024-01-16 17:44:05.9365271 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/multihead_attn/Unsqueeze
2024-01-16 17:44:05.9436674 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_13
2024-01-16 17:44:05.9512091 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_12
2024-01-16 17:44:05.9590938 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_11
2024-01-16 17:44:05.9669391 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_10
2024-01-16 17:44:05.9758009 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_6
2024-01-16 17:44:05.9837540 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze_3
2024-01-16 17:44:05.9909117 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.2/self_attn/Unsqueeze
2024-01-16 17:44:05.9989404 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze_17
2024-01-16 17:44:06.0060870 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze_16
2024-01-16 17:44:06.0144480 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze_15
2024-01-16 17:44:06.0217316 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze_14
2024-01-16 17:44:06.0299253 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze_9
2024-01-16 17:44:06.0379322 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/multihead_attn/Unsqueeze
2024-01-16 17:44:06.0459068 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_13
2024-01-16 17:44:06.0543823 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_12
2024-01-16 17:44:06.0628823 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_11
2024-01-16 17:44:06.0711186 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_10
2024-01-16 17:44:06.0795655 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_6
2024-01-16 17:44:06.0877730 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze_3
2024-01-16 17:44:06.0961331 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.1/self_attn/Unsqueeze
2024-01-16 17:44:06.1043870 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze_17
2024-01-16 17:44:06.1124994 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze_16
2024-01-16 17:44:06.1213028 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze_15
2024-01-16 17:44:06.1300441 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze_14
2024-01-16 17:44:06.1380419 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze
2024-01-16 17:44:06.1464922 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_13
2024-01-16 17:44:06.1544290 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_12
2024-01-16 17:44:06.1624733 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_11
2024-01-16 17:44:06.1711390 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_10
2024-01-16 17:44:06.1796458 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_3
2024-01-16 17:44:06.1885938 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/Unsqueeze
2024-01-16 17:44:06.1946146 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/Unsqueeze_1
2024-01-16 17:44:06.2009190 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze
2024-01-16 17:44:06.2102025 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/self_attn/Unsqueeze_6
2024-01-16 17:44:06.2178145 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/decoder/layers.0/multihead_attn/Unsqueeze_9
2024-01-16 17:44:06.2258868 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.5/self_attn/Unsqueeze_17
2024-01-16 17:44:06.2332930 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.5/self_attn/Unsqueeze_16
2024-01-16 17:44:06.2412123 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.5/self_attn/Unsqueeze_14
2024-01-16 17:44:06.2485259 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.5/self_attn/Unsqueeze_9
2024-01-16 17:44:06.2566488 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.4/self_attn/Unsqueeze_17
2024-01-16 17:44:06.2637376 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.4/self_attn/Unsqueeze_16
2024-01-16 17:44:06.2716915 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.4/self_attn/Unsqueeze_14
2024-01-16 17:44:06.2789598 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.4/self_attn/Unsqueeze_9
2024-01-16 17:44:06.2870689 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.3/self_attn/Unsqueeze_17
2024-01-16 17:44:06.2940611 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.3/self_attn/Unsqueeze_16
2024-01-16 17:44:06.3020849 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.3/self_attn/Unsqueeze_14
2024-01-16 17:44:06.3091170 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.3/self_attn/Unsqueeze_9
2024-01-16 17:44:06.3174981 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.2/self_attn/Unsqueeze_17
2024-01-16 17:44:06.3244424 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.2/self_attn/Unsqueeze_16
2024-01-16 17:44:06.3328650 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.2/self_attn/Unsqueeze_14
2024-01-16 17:44:06.3399581 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.2/self_attn/Unsqueeze_9
2024-01-16 17:44:06.3483070 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.1/self_attn/Unsqueeze_17
2024-01-16 17:44:06.3553263 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.1/self_attn/Unsqueeze_16
2024-01-16 17:44:06.3639431 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.1/self_attn/Unsqueeze_14
2024-01-16 17:44:06.3711855 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.1/self_attn/Unsqueeze_9
2024-01-16 17:44:06.3799603 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.0/self_attn/Unsqueeze_17
2024-01-16 17:44:06.3878715 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.0/self_attn/Unsqueeze_16
2024-01-16 17:44:06.3959246 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.0/self_attn/Unsqueeze_14
2024-01-16 17:44:06.4046978 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /transformer/encoder/layers.0/self_attn/Unsqueeze_9
2024-01-16 17:44:06.4131820 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_18
2024-01-16 17:44:06.4200199 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_16
2024-01-16 17:44:06.4267732 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_15
2024-01-16 17:44:06.4326282 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_14
2024-01-16 17:44:06.4383356 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_12
2024-01-16 17:44:06.4451086 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_11
2024-01-16 17:44:06.4543798 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_9
2024-01-16 17:44:06.4600426 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_8
2024-01-16 17:44:06.4661320 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_7
2024-01-16 17:44:06.4728330 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_5
2024-01-16 17:44:06.4785234 [W:onnxruntime:, unsqueeze_elimination.cc:20 onnxruntime::UnsqueezeElimination::Apply] UnsqueezeElimination cannot remove node /Unsqueeze_4

Basically unsqueeze cannot be removed for some reason... Why is that?

link to model: https://drive.google.com/file/d/1-itEFkapRXvYkKbknpFczqPFHyb3OrjK/view?usp=drive_link

Environment
TensorRT Version: 8.6

NVIDIA GPU: GTX 1660 Ti

NVIDIA Driver Version: 546.01

CUDA Version: 12.1

CUDNN Version: 8.9.7

Operating System: Winodws 10

Python Version (if applicable): 3.10.13

PyTorch Version (if applicable): 2.1.2+cu121

### To reproduce

polygraphy surgeon sanitize tomp101_head_latest3.onnx --fold-constants -o latest3_sanitized2.onnx

### Platform

Windows

### OS Version

10

### ONNX Runtime Installation

Built from Source

### ONNX Runtime Version or Commit ID

latest

### ONNX Runtime API

Python

### Architecture

X64

### Execution Provider

CUDA

**Comment by xadupre** [comment]
This error message lacks precision. It is returned when the following function fails: https://github.com/microsoft/onnxruntime/blob/main/onnxruntime/core/graph/graph_utils.cc#L541. The reason why a node cannot be replaced is one of the reasons mentioned in the function comments.
